core/test/integration/synchronization/ethereum_synchronization.cpp

```diff
@@ -0,0 +1,184 @@
+/*
+ *
+ * ethereum_synchronization
+ *
+ * Created by El Khalil Bellakrid on 27/07/2018.
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2018 Ledger
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+#include <gtest/gtest.h>
+#include "../BaseFixture.h"
+#include <set>
+#include <api/KeychainEngines.hpp>
+#include <utils/DateUtils.hpp>
+#include <wallet/ethereum/database/EthereumLikeAccountDatabaseHelper.h>
+#include <wallet/ethereum/transaction_builders/EthereumLikeTransactionBuilder.h>
+#include <wallet/ethereum/ERC20/ERC20LikeAccount.h>
+#include <iostream>
+using namespace std;
+
+class EthereumLikeWalletSynchronization : public BaseFixture {
+
+};
+
+TEST_F(EthereumLikeWalletSynchronization, MediumXpubSynchronization) {
+    auto pool = newDefaultPool();
+    {
+        auto configuration = DynamicObject::newInstance();
+        configuration->putString(api::Configuration::KEYCHAIN_ENGINE,api::KeychainEngines::BIP49_P2SH);
```

Comment:
So Ethereum uses BIP*? Interesting!

Reply:
Good catch, it uses the BIP44, this subject of keychain of ETH is more complicated than it seems, basically there is no standard on derivation of addresses on ETH (even the coin type sometimes is different from one wallet to another ...) we can talk about it if you need more details.
Here it's not even used (the test right after it was commented on it by the way)